osquery/tables/system/linux/md_tables.cpp

```diff
@@ -0,0 +1,733 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+#include <linux/major.h>
+#include <linux/raid/md_p.h>
+#include <linux/raid/md_u.h>
+
+#include <fstream>
+#include <memory>
+
+#include <boost/algorithm/string/trim.hpp>
+
+#include <osquery/core/conversions.h>
+#include <osquery/logger.h>
+
+#include "osquery/events/linux/udev.h"
+#include "osquery/tables/system/linux/md_tables.h"
+
+namespace osquery {
+namespace tables {
+
+std::string kMDStatPath = "/proc/mdstat";
+
+/**
+ * @brief Removes prefixing and suffixing character from each string in vector
+ *
+ * @param strs reference to vector of target strings
+ * @param c character to remove
+ *
+ */
+void trimStrs(std::vector<std::string>& strs) {
+  for (auto& s : strs) {
+    boost::algorithm::trim(s);
+  }
+}
+
+/**
+ * @brief reads mdstat file and parses as vector of lines
+ *
+ * @param lines reference to vector of strings to store lines in
+ *
+ * Trims all white space in lines and ignores empty lines (i.e. only spaces,
+ * tabs, etc.)
+ */
+inline void getLines(std::vector<std::string>& lines) {
+  std::ifstream handle(kMDStatPath);
+
+  std::string line;
+  if (handle.is_open()) {
+    while (getline(handle, line)) {
+      boost::algorithm::trim(line);
```

Comment:
One spot where it might be good to make a call on using your `trimStrs` defined above, or just stick with using `boost::algorithm::trim` everywhere. You might also `namespace alg = boost::algorithm;` above and then you could `alg::trim`?